File: leetcode/0566-reshape-the-matrix/solution.py
from typing import List


class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        m, n = len(mat), len(mat[0])
        if r * c != m * n:
            return mat

        result = [[0] * c for _ in range(r)]
        for k in range(m * n):
            result[k // c][k % c] = mat[k // n][k % n]

        return result

File: leetcode/0566-reshape-the-matrix/solution.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& mat, int r, int c) {
        int m = (int)mat.size(), n = (int)mat[0].size();
        if (r * c != m * n) {
            return mat;
        }

        vector<vector<int>> result(r, vector<int>(c, 0));
        for (int k = 0; k < m * n; k++) {
            result[k / c][k % c] = mat[k / n][k % n];
        }

        return result;
    }
};
